dev/com.ibm.ws.microprofile.openapi_test/test/com/ibm/ws/microprofile/openapi/validation/test/LinkValidatorTest.java

```diff
@@ -0,0 +1,222 @@
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.ibm.ws.microprofile.openapi.impl.model.OpenAPIImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.OperationImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.PathItemImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.PathsImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.links.LinkImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.responses.APIResponseImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.responses.APIResponsesImpl;
+import com.ibm.ws.microprofile.openapi.impl.validation.LinkValidator;
+import com.ibm.ws.microprofile.openapi.impl.validation.OperationValidator;
+import com.ibm.ws.microprofile.openapi.test.utils.TestValidationContextHelper;
+import com.ibm.ws.microprofile.openapi.test.utils.TestValidationHelper;
+import com.ibm.ws.microprofile.openapi.utils.OpenAPIModelWalker.Context;
+
+/**
+ *
+ */
+public class LinkValidatorTest {
+
+    String key = null;
+
+    OpenAPIImpl model = new OpenAPIImpl();
+    Context context = new TestValidationContextHelper(model);
+
+    @Before
+    public void setUpPaths() {
+        String pathNameOne = "/my-test-path-one";
+        PathItemImpl pathItemOne = new PathItemImpl(); //pathItem
+        OperationImpl getPathItemOne = new OperationImpl(); //operation
+        APIResponsesImpl responsesOne = new APIResponsesImpl(); //adding responses to operation
+        APIResponseImpl responseOne = new APIResponseImpl();
+        responsesOne.addApiResponse("200", responseOne.description("Operation successful"));
+        getPathItemOne.operationId("pathItemOneGetId").responses(responsesOne); //adding op ID and responses to operation
+        pathItemOne.setGET(getPathItemOne); //set operation of pathItem
+
+        PathItemImpl pathItemTwo = new PathItemImpl(); //pathItem
+        String pathNameTwo = "/my-test-path-two";
+        OperationImpl getPathItemTwo = new OperationImpl(); //operation
+        APIResponsesImpl responsesTwo = new APIResponsesImpl();
+        APIResponseImpl responseTwo = new APIResponseImpl();
+        responsesTwo.addApiResponse("400", responseTwo.description("Unable to complete operation"));
+        getPathItemTwo.operationId("pathItemTwoGetId").responses(responsesTwo); //adding op ID and responses to operation
+        pathItemTwo.setGET(getPathItemTwo); //set operation of pathItem
+
+        PathsImpl paths = new PathsImpl(); //paths
+        paths.addPathItem(pathNameOne, pathItemOne);
+        paths.addPathItem(pathNameTwo, pathItemTwo);
+
+        model.setPaths(paths);
+    }
+
+    @Test
+    public void testCorrectLinkWithRef() {
+        LinkValidator validator = LinkValidator.getInstance();
+        TestValidationHelper vh = new TestValidationHelper();
+
+        LinkImpl correctLink = new LinkImpl();
+        correctLink.setDescription("This is a correct test link");
+        String operationRef = "#/paths/my-test-path-two/get";
```

Comment:
this should be:
String operationRef = "#/paths/~1my-test-path-two/get";

The slashes in path should be escaped with ~1